src/include/duckdb/common/bitpacking.hpp

```diff
@@ -178,7 +178,31 @@ class BitpackingPrimitives {
 	}
 
 	template <class T>
-	static void UnPackGroup(data_ptr_t dst, data_ptr_t src, bitpacking_width_t width,
+	static void ApplyFrameOfReference(data_ptr_t dst, T frame_of_reference) {
+		if (!frame_of_reference) {
+			return;
+		}
+		for (idx_t i = 0; i < BITPACKING_ALGORITHM_GROUP_SIZE; i++) {
+			T value = Load<T>(dst + (i * sizeof(T)));
+			value += frame_of_reference;
+			Store<T>(value, dst + (i * sizeof(T)));
+		}
+	}
+
+	// Sign bit extension
+	template <class T, class T_U = typename std::make_unsigned<T>::type>
+	static void SignExtend(data_ptr_t dst, bitpacking_width_t width) {
+		T const mask = ((T_U)1) << (width - 1);
+		for (idx_t i = 0; i < BitpackingPrimitives::BITPACKING_ALGORITHM_GROUP_SIZE; ++i) {
+			T value = Load<T>(dst + i * sizeof(T));
+			value = value & ((((T_U)1) << width) - ((T_U)1));
+			T result = (value ^ mask) - mask;
+			Store(result, dst + i * sizeof(T));
+		}
+	}
+
+	template <class T>
+	static void UnPackGroup(data_ptr_t dst, data_ptr_t src, bitpacking_width_t width, T frame_of_reference,
```

Comment:
This function is also used in places where we don't currently do FOR i think, it would be nice to not have the overhead of applying 0 there

Reply:
I was experiencing a bug caused by addition on random memory (because of the algorithm group size) so I moved the applying of the frame of reference out of here, which should also fix this issue